src/hooks/useDydxClient.tsx

```diff
@@ -122,10 +123,17 @@ const useDydxClientContext = () => {
     defaultValue: SelectedGasDenom.USDC,
   });
 
+  useEffect(() => {
+    if (compositeClient) {
+      setSelectedGasDenom(gasDenom);
+    }
+  }, [compositeClient]);
```

Comment:
might as well add the eslint ignore since this is disobeying rules of hooks on purpose

Reply:
Also is it intended that changing network will set selected gas denom on the new client after it's recreated?

Reply:
Yep! That is intended. It is because the way we are storing this specific userPreference isn't network sepcific.